homeassistant/components/openhome/media_player.py

```diff
@@ -20,35 +21,45 @@
     SUPPORT_VOLUME_STEP,
 )
 from homeassistant.const import STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING
+from homeassistant.helpers import config_validation as cv, entity_platform
+
+from .const import ATTR_PIN_INDEX, DATA_OPENHOME, SERVICE_INVOKE_PIN
 
 SUPPORT_OPENHOME = SUPPORT_SELECT_SOURCE | SUPPORT_TURN_OFF | SUPPORT_TURN_ON
 
 _LOGGER = logging.getLogger(__name__)
 
-DEVICES = []
-
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Openhome platform."""
 
     if not discovery_info:
-        return True
+        return
+
+    openhome_data = hass.data.setdefault(DATA_OPENHOME, [])
 
     name = discovery_info.get("name")
     description = discovery_info.get("ssdp_description")
+
     _LOGGER.info("Openhome device found: %s", name)
-    device = Device(description)
+    device = await hass.async_add_executor_job(Device, description)
 
     # if device has already been discovered
-    if device.Uuid() in [x.unique_id for x in DEVICES]:
+    if device.Uuid() in [x.unique_id for x in openhome_data]:
```

Comment:
We can simplify the check here with the suggested change.

Reply:
Much more elegant, suspect I'd copied this from elsewhere. I've pushed some changes through.